Ui/Component/Listing/Column/TrackNumber.php

```diff
@@ -1,46 +1,60 @@
 <?php
-/**
- * Show the status of the track
- *
- * If you want to add improvements, please create a fork in our GitHub:
- * https://github.com/myparcelnl
- *
- * @author      Reindert Vetter <reindert@myparcel.nl>
- * @copyright   2010-2017 MyParcel
- * @license     http://creativecommons.org/licenses/by-nc-nd/3.0/nl/deed.en_US  CC BY-NC-ND 3.0 NL
- * @link        https://github.com/myparcelnl/magento
- * @since       File available since Release v0.1.0
- */
 
 namespace MyParcelNL\Magento\Ui\Component\Listing\Column;
 
-use \Magento\Sales\Api\OrderRepositoryInterface;
-use \Magento\Framework\View\Element\UiComponent\ContextInterface;
-use \Magento\Framework\View\Element\UiComponentFactory;
 use Magento\Sales\Model\Order;
-use \Magento\Ui\Component\Listing\Columns\Column;
-use \Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Ui\Component\Listing\Columns\Column;
+use MyparcelNL\Sdk\src\Helper\TrackTraceUrl;
 
 class TrackNumber extends Column
 {
+    /**
+     * Script tag to unbind the click event from the td wrapping the barcode link.
+     */
+    const SCRIPT_UNBIND_CLICK = "<script type='text/javascript'>jQuery('.myparcel-barcode-link').closest('td').unbind('click');</script>";
+
     /**
      * Set column MyParcel barcode to order grid
      *
      * @param array $dataSource
      *
      * @return array
      */
-    public function prepareDataSource(array $dataSource)
+    public function prepareDataSource(array $dataSource): array
     {
+        parent::prepareDataSource($dataSource);
+
+        if (! isset($dataSource['data']['items'])) {
+            return $dataSource;
+        }
+
         /**
-         * @var Order $order
+         * @var Order                  $order
          * @var Order\Shipment\Track[] $tracks
          */
-        if (isset($dataSource['data']['items'])) {
-            foreach ($dataSource['data']['items'] as & $item) {
-                if (key_exists('track_number', $item)) {
-                    $item[$this->getData('name')] = $item['track_number'];
-                }
+        foreach ($dataSource['data']['items'] as & $item) {
+            $addressParts = explode(",", $item['shipping_address']);
+
+            if (count($addressParts) === 4) {
+                [$company, $street, $city, $postalCode] = $addressParts;
+            } else {
+                [$street, $city, $postalCode] = $addressParts;
+            }
+
+            // Stop if either the barcode/status or postal code is missing.
+            if (! $item['track_number'] || ! $postalCode) {
+                continue;
+            }
+
+            $trackNumber = $item['track_number'];
+            $data = $this->getData('name');
+
+            // Only render the T&T as a link and add the script to remove the click handler if it's actually a barcode.
+            if (strpos($trackNumber, '3S') === 0) {
```

Comment:
```suggestion
            if (! empty($trackNumber) && $trackNumber != 'Concept') {
```

Reply:
Dat wordt erboven al afgevangen!
```php
if (! $item['track_number'] || ! $postalCode) {
    continue;
}
```

En het is nu "–" in plaats van "Concept" 😏